float64[] T
---
float64[] q_sols
int32 num_sols